picard/webservice/api_helpers.py

```diff
@@ -59,91 +58,74 @@ def _wrap_xml_metadata(data):
 
 
 class APIHelper(object):
+    base_path = "/"
 
-    def __init__(self, host, port, api_path, webservice):
-        self._host = host
-        self._port = port
-        self.api_path = api_path
+    def __init__(self, webservice):
         self._webservice = webservice
 
     @property
     def webservice(self):
         return self._webservice
 
     @property
-    def host(self):
-        return self._host
-
-    @property
-    def port(self):
-        return self._port
-
-    def get(self, path_list, handler, priority=False, important=False, mblogin=False,
-            cacheloadcontrol=None, refresh=False, queryargs=None, parse_response_type=DEFAULT_RESPONSE_PARSER_TYPE):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.get(self.host, self.port, path, handler,
-                                    priority=priority, important=important, mblogin=mblogin,
-                                    refresh=refresh, queryargs=queryargs, parse_response_type=parse_response_type)
-
-    def post(self, path_list, data, handler, priority=False, important=False,
-             mblogin=True, queryargs=None, parse_response_type=DEFAULT_RESPONSE_PARSER_TYPE,
-             request_mimetype=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.post(self.host, self.port, path, data, handler,
-                                     priority=priority, important=important, mblogin=mblogin,
-                                     queryargs=queryargs, parse_response_type=parse_response_type,
-                                     request_mimetype=request_mimetype)
-
-    def put(self, path_list, data, handler, priority=True, important=False,
-            mblogin=True, queryargs=None, request_mimetype=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.put(self.host, self.port, path, data, handler,
-                                    priority=priority, important=important, mblogin=mblogin,
-                                    queryargs=queryargs, request_mimetype=request_mimetype)
-
-    def delete(self, path_list, handler, priority=True, important=False,
-               mblogin=True, queryargs=None):
-        path = self.api_path + "/".join(path_list)
-        return self._webservice.delete(self.host, self.port, path, handler,
-                                       priority=priority, important=important, mblogin=mblogin,
-                                       queryargs=queryargs)
+    def url(self):
+        raise NotImplementedError
+
+    def url_from_path_list(self, path_list):
+        url = self.url
+        url.setPath("/".join([self.base_path] + list(path_list)))
+        return url
+
+    def get(self, path_list, handler, **kwargs):
```

Comment:
I am not too sure on the `path_list` . Is this really the most convenient interface? Taking my LB API example from above:

```python
lb_api = APIHelper(ws, base_url="https://api.listenbrainz.org/1/")
lb_api.get(('user', 'outsidecontext', 'playing-now'), some_handler)
```

vs.

```python
lb_api = APIHelper(ws, base_url="https://api.listenbrainz.org/1/")
lb_api.get('user/outsidecontext/playing-now', some_handler)
```

To me pathing the path as a string seems easier to read and write. Or maybe be flexible?

Reply:
Well, I kept it because it was there already.
I was wondering about that too.

Reply:
Check https://github.com/metabrainz/picard/pull/2225/commits/fb6401cda297b4a0c8b658dfcbe45de7a0daab05

I think that's simpler and easier to understand.